Sources/SwiftProtobuf/_MessageStorage.swift

```diff
@@ -235,22 +239,14 @@ extension _MessageStorage {
     /// Copy-initializes the field associated with the given layout information in the destination
     /// storage using its value from this storage.
     public func copyField<T>(_ field: FieldLayout, to destination: _MessageStorage, type: T.Type) {
-        switch field.presence {
-        case .oneOfMember:
-            // TODO: Support oneof fields.
-            break
+        guard isPresent(field) else { return }
```

Comment:
Worth an any asset to confirm we only call this on arrays or non pod fields?

Reply:
This is called on all types of fields. Even for POD fields that don't have explicit presence, we use has-bits because they're cheap and avoid having different code paths.

Reply:
Chatted offline, this was about adding a check as the first line of copyFields, not about isPresent. Tony's going to consider options for adding assertions/invariant checks in some way.